tests/test_qp.py

```diff
@@ -371,3 +301,30 @@ def get_qp_init_attr_ex(cq, pd, attr, attr_ex, qpt):
     qia.recv_cq = cq
     qia.pd = pd  # Only XRCD can be created without a PD
     return qia
+
+
+def get_qp_type_str(qp_type):
+    qpt_d = {
+        e.IBV_QPT_RC : 'RC',
+        e.IBV_QPT_UC : 'UC',
+        e.IBV_QPT_UD : 'UD',
+        e.IBV_QPT_RAW_PACKET : 'RAW Packet',
+        e.IBV_QPT_XRC_SEND : 'XRC Send',
+        e.IBV_QPT_XRC_RECV : 'XRC Recv',
+        e.IBV_QPT_DRIVER : 'Driver'
+    }
+    return qpt_d.get(qp_type)
+
+
+def get_qp_state_str(qp_state):
+    qps_d = {
+        e.IBV_QPS_RESET : 'RESET',
+        e.IBV_QPS_INIT : 'INIT',
+        e.IBV_QPS_RTR : 'RTR',
+        e.IBV_QPS_RTS : 'RTS',
+        e.IBV_QPS_SQD : 'SQD',
+        e.IBV_QPS_SQE : 'SQE',
+        e.IBV_QPS_ERR : 'ERR',
+        e.IBV_QPS_UNKNOWN : 'UNKNOWN'
+    }
+    return qps_d.get(qp_state)
```

Comment:
You can use qp_type_to_str and qp_state_to_str respectively from pyverbs.utils

Reply:
I missed these two functions, Fixed.